api/src/wfl/api/workloads.clj

```diff
@@ -182,6 +186,11 @@
              :pipeline pipeline
              :type     ::invalid-pipeline})))
 
+(defmethod to-log
```

Comment:
Example log for a legacy workload:

```{"message":{"column":5,"namespace":"wfl.server","result":"Updating workload","workload":{"uuid":"dcc56c2d-ca80-44e0-919d-f152467d39fa","pipeline":"GDCWholeGenomeSomaticSingleSample","project":"(Test) okotsopo/GH-1617-workload-source-logs","labels":[]}},"severity":"INFO","logging.googleapis.com/sourceLocation":{"file":"wfl/server.clj","function":"\"Updating workload\"","line":98},"time":"2022-02-17T20:18:50.571237Z"}```

Reply:
> But why not just log the workload?

Good question.  Don't know about the legacies, but a loaded stage workload object can be several hundred lines.  Here's a local example.

```{:started #inst"2022-02-10T14:20:58.050434000-00:00",
 :watchers [["email" "hornet@broadinstitute.org"] ["slack" "C026PTM4XPA" "#hornet-slack-app-testing"]],
 :labels ["hornet:test" "project:(Test) rfricke/GH-1566-workloadinfologs"],
 :creator "wfl-non-prod@broad-gotc-dev.iam.gserviceaccount.com",
 :updated #inst"2022-02-10T14:55:36.670933000-00:00",
 :type :workload,
 :created #inst"2022-02-10T14:20:38.325012000-00:00",
 :source {:pollingIntervalMinutes 1,
          :last_checked #inst"2022-02-10T14:54:12.349939000-00:00",
          :snapshotReaders ["hornet@firecloud.org"],
          :table "inputs",
          :type "TerraDataRepoSource",
          :stopped #inst"2022-02-10T14:55:34.037762000-00:00",
          :details "TerraDataRepoSource_000001107",
          :id 106,
          :loadTag "loadTagToMonitor",
          :dataset {:description "test inputs and outputs for the illumina_genotyping_array pipeline",
                    :schema {:tables [{:name "outputs",
                                       :columns [{:name "arrays_variant_calling_control_metrics",
                                                  :datatype "fileref",
                                                  :array_of false}
                                                 {:name "arrays_variant_calling_detail_metrics",
                                                  :datatype "fileref",
                                                  :array_of false}
                                                 {:name "arrays_variant_calling_summary_metrics",
                                                  :datatype "fileref",
                                                  :array_of false}
                                                 {:name "contamination_metrics", :datatype "fileref", :array_of false}
                                                 {:name "genotype_concordance_contingency_metrics",
                                                  :datatype "fileref",
                                                  :array_of false}
                                                 {:name "genotype_concordance_detail_metrics",
                                                  :datatype "fileref",
                                                  :array_of false}
                                                 {:name "genotype_concordance_failed",
                                                  :datatype "boolean",
                                                  :array_of false}
                                                 {:name "genotype_concordance_summary_metrics",
                                                  :datatype "fileref",
                                                  :array_of false}
                                                 {:name "green_idat_md5_cloud_path",
                                                  :datatype "fileref",
                                                  :array_of false}
                                                 {:name "gtc", :datatype "fileref", :array_of false}
                                                 {:name "output_fingerprint_vcf", :datatype "fileref", :array_of false}
                                                 {:name "output_fingerprint_vcf_index",
                                                  :datatype "fileref",
                                                  :array_of false}
                                                 {:name "output_vcf", :datatype "fileref", :array_of false}
                                                 {:name "output_vcf_index", :datatype "fileref", :array_of false}
                                                 {:name "output_vcf_md5_cloud_path",
                                                  :datatype "fileref",
                                                  :array_of false}
                                                 {:name "red_idat_md5_cloud_path", :datatype "fileref", :array_of false}],
                                       :primaryKey ["arrays_variant_calling_control_metrics"
                                                    "arrays_variant_calling_detail_metrics"
                                                    "arrays_variant_calling_summary_metrics"
                                                    "contamination_metrics"
                                                    "genotype_concordance_contingency_metrics"
                                                    "genotype_concordance_detail_metrics"
                                                    "genotype_concordance_failed"
                                                    "genotype_concordance_summary_metrics"
                                                    "green_idat_md5_cloud_path"
                                                    "gtc"
                                                    "output_fingerprint_vcf"
                                                    "output_fingerprint_vcf_index"
                                                    "output_vcf"
                                                    "output_vcf_index"
                                                    "output_vcf_md5_cloud_path"
                                                    "red_idat_md5_cloud_path"],
                                       :partitionMode "date",
                                       :datePartitionOptions {:column "datarepo_ingest_date"},
                                       :intPartitionOptions nil,
                                       :rowCount nil}
                                      {:name "inputs",
                                       :columns [{:name "analysis_version_number", :datatype "integer", :array_of false}
                                                 {:name "chip_well_barcode", :datatype "string", :array_of false}
                                                 {:name "green_idat_cloud_path", :datatype "fileref", :array_of false}
                                                 {:name "red_idat_cloud_path", :datatype "fileref", :array_of false}
                                                 {:name "sample_alias", :datatype "string", :array_of false}
                                                 {:name "reported_gender", :datatype "string", :array_of false}
                                                 {:name "ingested", :datatype "timestamp", :array_of false}],
                                       :primaryKey ["analysis_version_number"
                                                    "chip_well_barcode"
                                                    "green_idat_cloud_path"
                                                    "red_idat_cloud_path"
                                                    "sample_alias"
                                                    "reported_gender"
                                                    "ingested"],
                                       :partitionMode "date",
                                       :datePartitionOptions {:column "datarepo_ingest_date"},
                                       :intPartitionOptions nil,
                                       :rowCount nil}],
                             :relationships [],
                             :assets []},
                    :dataProject "datarepo-5777e368",
                    :defaultProfileId "6370f5a1-d777-4991-8200-ceab83521d43",
                    :secureMonitoringEnabled false,
                    :name "illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96",
                    :accessInformation {:bigQuery {:datasetName "datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96",
                                                   :datasetId "datarepo-5777e368:datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96",
                                                   :projectId "datarepo-5777e368",
                                                   :link "https://console.cloud.google.com/bigquery?project=datarepo-5777e368&ws=!datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96&d=datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96&p=datarepo-5777e368&page=dataset",
                                                   :tables [{:name "outputs",
                                                             :id "datarepo-5777e368:datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96.outputs",
                                                             :qualifiedName "datarepo-5777e368.datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96.outputs",
                                                             :link "https://console.cloud.google.com/bigquery?project=datarepo-5777e368&ws=!datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96&d=datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96&p=datarepo-5777e368&page=table&t=outputs",
                                                             :sampleQuery "SELECT * FROM `datarepo-5777e368.datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96.outputs` LIMIT 1000"}
                                                            {:name "inputs",
                                                             :id "datarepo-5777e368:datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96.inputs",
                                                             :qualifiedName "datarepo-5777e368.datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96.inputs",
                                                             :link "https://console.cloud.google.com/bigquery?project=datarepo-5777e368&ws=!datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96&d=datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96&p=datarepo-5777e368&page=table&t=inputs",
                                                             :sampleQuery "SELECT * FROM `datarepo-5777e368.datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96.inputs` LIMIT 1000"}]},
                                        :parquet nil},
                    :id "ef2ba069-a8c5-4a57-8444-b1567a3cec0a",
                    :storage [{:region "us-central1", :cloudResource "bigquery", :cloudPlatform "gcp"}
                              {:region "us-east4", :cloudResource "firestore", :cloudPlatform "gcp"}
                              {:region "us-central1", :cloudResource "bucket", :cloudPlatform "gcp"}],
                    :createdDate "2022-02-10T14:19:39.867650Z",
                    :defaultSnapshotId nil}},
 :finished #inst"2022-02-10T00:35:00.000000000-00:00",
 :stopped #inst"2022-02-10T14:55:34.037149000-00:00",
 :id 1107,
 :commit "6caf37cb67d0361c237eb2f083270293ec944854",
 :uuid "e66439f4-cb6d-4d8b-93e8-03a1c309dcd6",
 :executor {:id 111,
            :details "TerraExecutor_000001107",
            :type "TerraExecutor",
            :workspace "wfl-dev/Illumina-Genotyping-Array-Templateecf09d017b8d4033bab8d5feeae86987",
            :methodConfiguration "warp-pipelines/IlluminaGenotypingArray",
            :methodConfigurationVersion 2,
            :fromSource "importSnapshot"},
 :version "0.13.0",
 :sink {:id 47,
        :details "TerraDataRepoSink_000001107",
        :dataset {:description "test inputs and outputs for the illumina_genotyping_array pipeline",
                  :schema {:tables [{:name "outputs",
                                     :columns [{:name "arrays_variant_calling_control_metrics",
                                                :datatype "fileref",
                                                :array_of false}
                                               {:name "arrays_variant_calling_detail_metrics",
                                                :datatype "fileref",
                                                :array_of false}
                                               {:name "arrays_variant_calling_summary_metrics",
                                                :datatype "fileref",
                                                :array_of false}
                                               {:name "contamination_metrics", :datatype "fileref", :array_of false}
                                               {:name "genotype_concordance_contingency_metrics",
                                                :datatype "fileref",
                                                :array_of false}
                                               {:name "genotype_concordance_detail_metrics",
                                                :datatype "fileref",
                                                :array_of false}
                                               {:name "genotype_concordance_failed",
                                                :datatype "boolean",
                                                :array_of false}
                                               {:name "genotype_concordance_summary_metrics",
                                                :datatype "fileref",
                                                :array_of false}
                                               {:name "green_idat_md5_cloud_path", :datatype "fileref", :array_of false}
                                               {:name "gtc", :datatype "fileref", :array_of false}
                                               {:name "output_fingerprint_vcf", :datatype "fileref", :array_of false}
                                               {:name "output_fingerprint_vcf_index",
                                                :datatype "fileref",
                                                :array_of false}
                                               {:name "output_vcf", :datatype "fileref", :array_of false}
                                               {:name "output_vcf_index", :datatype "fileref", :array_of false}
                                               {:name "output_vcf_md5_cloud_path", :datatype "fileref", :array_of false}
                                               {:name "red_idat_md5_cloud_path", :datatype "fileref", :array_of false}],
                                     :primaryKey ["arrays_variant_calling_control_metrics"
                                                  "arrays_variant_calling_detail_metrics"
                                                  "arrays_variant_calling_summary_metrics"
                                                  "contamination_metrics"
                                                  "genotype_concordance_contingency_metrics"
                                                  "genotype_concordance_detail_metrics"
                                                  "genotype_concordance_failed"
                                                  "genotype_concordance_summary_metrics"
                                                  "green_idat_md5_cloud_path"
                                                  "gtc"
                                                  "output_fingerprint_vcf"
                                                  "output_fingerprint_vcf_index"
                                                  "output_vcf"
                                                  "output_vcf_index"
                                                  "output_vcf_md5_cloud_path"
                                                  "red_idat_md5_cloud_path"],
                                     :partitionMode "date",
                                     :datePartitionOptions {:column "datarepo_ingest_date"},
                                     :intPartitionOptions nil,
                                     :rowCount nil}
                                    {:name "inputs",
                                     :columns [{:name "analysis_version_number", :datatype "integer", :array_of false}
                                               {:name "chip_well_barcode", :datatype "string", :array_of false}
                                               {:name "green_idat_cloud_path", :datatype "fileref", :array_of false}
                                               {:name "red_idat_cloud_path", :datatype "fileref", :array_of false}
                                               {:name "sample_alias", :datatype "string", :array_of false}
                                               {:name "reported_gender", :datatype "string", :array_of false}
                                               {:name "ingested", :datatype "timestamp", :array_of false}],
                                     :primaryKey ["analysis_version_number"
                                                  "chip_well_barcode"
                                                  "green_idat_cloud_path"
                                                  "red_idat_cloud_path"
                                                  "sample_alias"
                                                  "reported_gender"
                                                  "ingested"],
                                     :partitionMode "date",
                                     :datePartitionOptions {:column "datarepo_ingest_date"},
                                     :intPartitionOptions nil,
                                     :rowCount nil}],
                           :relationships [],
                           :assets []},
                  :dataProject "datarepo-5777e368",
                  :defaultProfileId "6370f5a1-d777-4991-8200-ceab83521d43",
                  :secureMonitoringEnabled false,
                  :name "illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96",
                  :accessInformation {:bigQuery {:datasetName "datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96",
                                                 :datasetId "datarepo-5777e368:datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96",
                                                 :projectId "datarepo-5777e368",
                                                 :link "https://console.cloud.google.com/bigquery?project=datarepo-5777e368&ws=!datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96&d=datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96&p=datarepo-5777e368&page=dataset",
                                                 :tables [{:name "outputs",
                                                           :id "datarepo-5777e368:datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96.outputs",
                                                           :qualifiedName "datarepo-5777e368.datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96.outputs",
                                                           :link "https://console.cloud.google.com/bigquery?project=datarepo-5777e368&ws=!datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96&d=datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96&p=datarepo-5777e368&page=table&t=outputs",
                                                           :sampleQuery "SELECT * FROM `datarepo-5777e368.datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96.outputs` LIMIT 1000"}
                                                          {:name "inputs",
                                                           :id "datarepo-5777e368:datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96.inputs",
                                                           :qualifiedName "datarepo-5777e368.datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96.inputs",
                                                           :link "https://console.cloud.google.com/bigquery?project=datarepo-5777e368&ws=!datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96&d=datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96&p=datarepo-5777e368&page=table&t=inputs",
                                                           :sampleQuery "SELECT * FROM `datarepo-5777e368.datarepo_illumina_genotyping_array85808ce23f7a41298ed1c7dcf5cc7c96.inputs` LIMIT 1000"}]},
                                      :parquet nil},
                  :id "ef2ba069-a8c5-4a57-8444-b1567a3cec0a",
                  :storage [{:region "us-central1", :cloudResource "bigquery", :cloudPlatform "gcp"}
                            {:region "us-east4", :cloudResource "firestore", :cloudPlatform "gcp"}
                            {:region "us-central1", :cloudResource "bucket", :cloudPlatform "gcp"}],
                  :createdDate "2022-02-10T14:19:39.867650Z",
                  :defaultSnapshotId nil},
        :table "outputs",
        :fromOutputs {:output_vcf_index "output_vcf_index",
                      :gtc "gtc",
                      :arrays_variant_calling_detail_metrics "arrays_variant_calling_detail_metrics",
                      :green_idat_md5_cloud_path "green_idat_md5_cloud_path",
                      :output_fingerprint_vcf_index "output_fingerprint_vcf_index",
                      :contamination_metrics "contamination_metrics",
                      :output_fingerprint_vcf "output_fingerprint_vcf",
                      :red_idat_md5_cloud_path "red_idat_md5_cloud_path",
                      :genotype_concordance_failed "genotype_concordance_failed",
                      :output_vcf "output_vcf",
                      :genotype_concordance_summary_metrics "genotype_concordance_summary_metrics",
                      :arrays_variant_calling_control_metrics "arrays_variant_calling_control_metrics",
                      :genotype_concordance_detail_metrics "genotype_concordance_detail_metrics",
                      :output_vcf_md5_cloud_path "output_vcf_md5_cloud_path",
                      :arrays_variant_calling_summary_metrics "arrays_variant_calling_summary_metrics",
                      :genotype_concordance_contingency_metrics "genotype_concordance_contingency_metrics"},
        :type "TerraDataRepoSink"}}```

Reply:
Well … then … don't do *that*.